dbt/include/trino/macros/materializations/table.sql

```diff
@@ -1,58 +1,27 @@
-{% materialization table, adapter='trino' %}
+{% materialization table, adapter = 'trino' %}
   {%- set identifier = model['alias'] -%}
-  {%- set tmp_identifier = model['name'] + '__dbt_tmp' -%}
-  {%- set backup_identifier = model['name'] + '__dbt_backup' -%}
 
   {%- set old_relation = adapter.get_relation(database=database, schema=schema, identifier=identifier) -%}
   {%- set target_relation = api.Relation.create(identifier=identifier,
                                                 schema=schema,
                                                 database=database,
                                                 type='table') -%}
-  {%- set intermediate_relation = api.Relation.create(identifier=tmp_identifier,
-                                                      schema=schema,
-                                                      database=database,
-                                                      type='table') -%}
 
-  {%- set backup_relation_type = 'table' if old_relation is none else old_relation.type -%}
-  {%- set backup_relation = api.Relation.create(identifier=backup_identifier,
-                                                schema=schema,
-                                                database=database,
-                                                type=backup_relation_type) -%}
-
-  {%- set exists_as_table = (old_relation is not none and old_relation.is_table) -%}
-  {%- set exists_as_view = (old_relation is not none and old_relation.is_view) -%}
-
-
-  -- drop the temp relations if they exists for some reason
-  {{ adapter.drop_relation(intermediate_relation) }}
-  {{ adapter.drop_relation(backup_relation) }}
-
-  {{ run_hooks(pre_hooks, inside_transaction=False) }}
+  {{ run_hooks(pre_hooks) }}
 
-  -- `BEGIN` happens here:
-  {{ run_hooks(pre_hooks, inside_transaction=True) }}
+  -- cleanup
+  {%- if old_relation is not none -%}
+      {{ adapter.drop_relation(old_relation) }}
+  {%- endif -%}
 
   -- build model
   {% call statement('main') -%}
-    {{ create_table_as(False, intermediate_relation, sql) }}
+    {{ create_table_as(False, target_relation, sql) }}
   {%- endcall %}
 
-  -- cleanup
-  {% if old_relation is not none %}
-      {{ adapter.rename_relation(old_relation, backup_relation) }}
-  {% endif %}
-
-  {{ adapter.rename_relation(intermediate_relation, target_relation) }}
```

Comment:
The main advantage of doing the  intermediate relation renaming is that the `table` materialization appears to be atomic to the end user.

The change proposed in this PR however handles the `table` materialization in two steps:
- the first one drops the existing table
- the second one creates the table which corresponds to the newest model

Now if the execution flow is interrupted in between, the existing table  would be dropped and there would be no backup available.

I do know that AWS Glue doesn't support table renames and that dbt-trino would need to cope with this aspect. However we need to keep in mind that dbt-trino is a generic (and not AWS Glue specific) library.

Let's investigate a bit more how we could cover the need of dealing with AWS Glue tables in dbt-trino.

Reply:
We could verify whether table renames are supported (create a table , try to rename it and see whether an exception with `NOT_SUPPORTED`  error code is being returned) and fallback to this rather simplistic flow. However this is a rather big overhead because most of the databases actually do support table renames.

What about providing a specific materialization for aws-glue tables? This would be a reasonable compromise right?